Automattic/wp-calypso
# Add VBE editor injector

### Changes
This PR aims to create a standalone bundle that attaches the Gutenberg editor to a textarea and makes it support blocks. By standalone, I mean a library that exposes global and doesn't have any dependencies.

### Testing steps

#### Regression testing

1. Make sure the editor works in Calypso by heading to /comment and embedding a block or two. If one embed works, all will work, don't sweat it.


#### Testing in Verbum
1. Navigate to `packages/verbum-block-editor`.
2. Run `yarn dev --sync`.
3. In Verbum https://github.com/Automattic/verbum/pull/128, checkout `use/packaged-editor` and run `npm run start:sync`.
4. Sandbox widgets.wp.com.
5. Sandbox a site.
6. Visit a blogpost page on that site. 
7. The editor should and should work.
8. To verify it's the new editor, verify that embeds don't work. 

<details><summary>Old description</summary>
<p>
## Proposed Changes

This PR aims to create a standalone bundle that attaches the Gutenberg editor to a textarea and makes it support blocks. By standalone, I mean a library that exposes global and doesn't have any dependencies.

The goal is to be able to publish it to NPM, then import it in [here](https://github.com/Automattic/verbum/blob/trunk/src/components/comment-input-field.tsx) and use it.

### Issue
For some reason, the editor is not being rendered correctly. It's a bit hard to describe the issue, but if you follow the steps, and take a quick look at the code, you'll see it right away!

### Reproduction steps

1. please checkout the branch of this PR.
9. cd into packages/verbum-block-editor
10. run yarn build.
11. In the dist file, add the following HTML file.
12. Serve the dist folder using serve or whatever you like (`npm i serve --global && serve`).
13. Visit the served page, and in the console, run verbumBlockEditor.attachGutenberg(editor, console.log)
14. The rendering won’t work and you’ll get a white page.
15. Expected result is that you’d get a Gutenberg editor to replace the textarea.

I suspect it's a JSX issue.

Related thread: p1704811519896619-slack-C7YPUHBB2

```html
<!doctype html>
<html lang="en">
	<head>
		<meta name="description" content="Webpage description goes here" />
		<meta charset="utf-8" />
		<title>Change_me</title>
		<meta name="viewport" content="width=device-width, initial-scale=1" />
		<meta name="author" content="" />

		<link rel="stylesheet" href="main.css" />
	</head>

	<body>
		<div class="container">
			<textarea style="width: 500px; height: 300px" id="editor"></textarea>
		</div>

		<script type="module" src="./main.min.js"></script>
	</body>
</html>
```

</p>
</details>
